### PR TITLE
chore: Make header names that always need to be sent constants

### PR DESCRIPTION
**Pull request by poppoerika**
This is a follow-up pr for the implementation option 2 mentioned [here](https://github.com/momentohq/client-sdk-csharp/pull/37#discussion_r840714313).

### CODE REVIEW

File: Momento/HeaderInterceptor.cs [hunk 1]
```diff
@@ -2,11 +2,14 @@
 using Grpc.Core;
 using Grpc.Core.Interceptors;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace MomentoSdk
 {
     class Header
     {
+        private const string Agent = "Agent";
```
Comment (gautamomento) [inline Momento/HeaderInterceptor.cs:11]: I dont think we should add this in each file. Define once and reference in each class.

e.g. drop the constant private here
```
const string AGENT_KEY = "Agent";
const string AUTHORIZATION_KEY = "Authorization";
```

And then in DataGrpcManager reference in `Headers.AGENT_KEY`

Comment (gautamomento) [reply Momento/HeaderInterceptor.cs:11]: And just one note, you probably should looking up naming conventions for C# for constants `AGENT_KEY` is what is done in Java. So, check the appropriate C# guideline.

Comment (poppoerika) [reply Momento/HeaderInterceptor.cs:11]: It seems like `FullProductName` is its naming convention for constants.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/const

**Review by gautamomento — APPROVED** [review]
Thank you for doing this.

LGTM

**Pull request merged**